### DISCUSSION TITLE
关于 List 组件开发相关

### DISCUSSION THREAD

**Discussion by Tyh2001** [FightingDesign/fighting-design#100] [Ideas]
# List 列表

## 说明

分为两个组件 `f-list` 和 `f-list-item`

## 设计稿

https://www.figma.com/file/iPQwcLbuH3aFklm9IATT00/List-%E5%88%97%E8%A1%A8?node-id=0%3A1

## List Attributes

- `show-num` 是否展示编号
- `zebra` 是否展示斑马纹
- `hover-show` 是否在 `hover` 状态展示投影效果
- `important-list` 重点行，传入一个 `number[]` 类型。重点行会高亮
- `text-color` 字体颜色
- `border-color` 边框颜色
- `background` 背景色
- `size` 不同尺寸
- `split` 是否显示分割线
- `max-height` 最大高度，超出滚动
- `center` 是否居中

## List Slots

- `default` 默认信息 
- `header` 头部信息
- `footer` 页脚信息

## List-Item Attributes

——

## List-Item Slots

- `default` 默认信息 
- `option` 左侧选项信息

**Comment by Tyh2001** [comment]
`show-num` 未实现？
`hover-show` 未实现？
`maxHeight` 配置项有问题，参考下面代码：

```html
  <f-list maxHeight="100px">
    <f-list-item>你好啊这是一个列表</f-list-item>
    <f-list-item>你好啊这是一个列表</f-list-item>
    <f-list-item>你好啊这是一个列表</f-list-item>
    <f-list-item>你好啊这是一个列表</f-list-item>
    <f-list-item>你好啊这是一个列表</f-list-item>
  </f-list>
```

@HYzihong

**Reply by HYzihong** [reply]
maxHeight 这个我看见好像没啥问题（可能是没看到问题）
hover-show、show-num代码我还在努力~

**Comment by Tyh2001** [comment]
![daeb65aee42b4dd1fb6f6d60e5cade5](https://user-images.githubusercontent.com/73180970/181139388-f83f7492-af4c-41f9-b88c-a091cad8d9e7.jpg)
上图是 `maxHeight ` 的问题，会给每个 `f-list-item` 添加，应该是给 `f-list` 添加

`show-num` 就是显示编号，其实将 `f-list` 的 `ul` 标签转换成 `ol` 即可实现

`hover-show` 效果可以参考我之前写的组件库 [over 状态展示投影](https://tianyuhao.cn/v3/components/list.html#hover-%E7%8A%B6%E6%80%81%E5%B1%95%E7%A4%BA%E6%8A%95%E5%BD%B1)

@HYzihong

**Reply by HYzihong** [reply]
哦哦，我理解成每个item的max-height了，那我改成list的max-height，删除item的max-height就好了
hover效果我已经搞好了，其他的今天晚上可以搞定，我感觉我今天晚上可以更新这个list组件

**Reply by HYzihong** [reply]
split 是否显示分割线 这个我没有太理解要达到什么样子的效果，有时间跟我表达一下，哈哈

**Reply by Tyh2001** [reply]
split 就是 `f-list` 会显示一个大的边框，`f-list-item` 每个之间就没有边框了

**Reply by HYzihong** [reply]
好的，就是`f-list-item`的自己border没了，`f-list`显示border了

**Reply by Tyh2001** [reply]
没错~

**Reply by HYzihong** [reply]
`f-list-item`我觉得也可以默认自适应高度，配置`maxHeight `时，也设置`max-height`怎么样

**Reply by Tyh2001** [reply]
可以的。`f-list-item` 应该是自适应高度的

**Reply by HYzihong** [reply]
ok

**Comment by Tyh2001** [comment]
可能后面还会新增无序列表和有序列表，前面是数字或者点

**Reply by HYzihong** [reply]
成，我觉得可以

**Comment by Tyh2001** [comment]
最近差点忘记了这个组件，发现文档还都没写。。。。

然后组件内部我看了下，优化了非常多的地方，你可以看下

https://github.com/FightingDesign/fighting-design/commit/b7724c90adc50a48c3226e413d1547378fab8855

@HYzihong
